tests/test_transfer.py

```diff
@@ -63,6 +63,29 @@ def transfer(adlfs, src, dst, offset, size, blocksize, buffersize, shutdown_even
                                                for chunk in f.chunks])
 
 
+def test_update_progress(azure):
+    """
+    Upload a 32 bytes file in chunks of 8 and test that the progress is incrementally
+        updated
+    """
+    calls = []
+
+    def recording_callback(progress, total):
+        calls.append((progress, total))
+
+    def transfer(adlfs, src, dst, offset, size, blocksize, buffersize, shutdown_event=None):
+        time.sleep(0.01)
```

Comment:
There's a bug in the way tasks are submitted and how state is handled on the callback #177
The sleep is a workaround used throughout the tests to avoid a race condition throughout the tests that do `client.run()`. 
Initially I assumed there's an issue with how pytests runs the tests and kept the sleep. However on a second look I found the bug.
Since the bug is not introduced by this code,  I'd keep this as is and rely on having the fix as a subsequent commit.